tests/test_type_client.py

```diff
@@ -87,16 +87,36 @@ def test_single_assays(self):
             tc = TypeClient()
         except RuntimeError:
             tc = TypeClient(self.config['typeserviceurl'])
```

Comment:
I didn't get your comment above:
>  Any test run against the dummy client can also be run against the actual client

and maybe that's related to these lines? I really don't understand what's going on here. As I understand it, there is one testing class in `test_type_client.py`: `TestTypeClientUninitialized`. In that file, the test is run with the _real_ type client.... and then it's imported again and replaced with this mock in `test_dummy_type_client.py`.

Usually, you use mocks when the object in question is part of a bigger picture and you wouldn't want to use the real thing... perhaps it depends on network / outside database / user input / just non-deterministic. But since you seem to be running exactly the same tests against the real thing, and the mock, I really don't see what having the mock is buying you.

Reply:
> I didn't get your comment above:
> 
> > Any test run against the dummy client can also be run against the actual client
> 
> and maybe that's related to these lines? I really don't understand what's going on here. As I understand it, there is one testing class in `test_type_client.py`: `TestTypeClientUninitialized`. In that file, the test is run with the _real_ type client.... and then it's imported again and replaced with this mock in `test_dummy_type_client.py`.
> 
> Usually, you use mocks when the object in question is part of a bigger picture and you wouldn't want to use the real thing... perhaps it depends on network / outside database / user input / just non-deterministic. But since you seem to be running exactly the same tests against the real thing, and the mock, I really don't see what having the mock is buying you.

I think this is the heart of the issue.  There is a problem with implementing tests for this PR which *don't* use mocks because that test must depend on the companion PR to search-api.  For procedural reasons that PR can't be advanced to the test-release branch until it has been tested in its full context.  But search-api includes commons in its requirements.txt, and so the full context for testing includes *this* PR.  So there is sort of a circular dependency.

I'm trying to break that dependency by including the mock version of the test routine, but obviously I want to provide a test that actually works in the full context, which would reference the API in search-api rather than using the mock mechanism.  So I write a set of tests that works with the full context, matching the output of the search-api.  

But when I set up the mock for that test, I want to use the same test cases for the mock and non-mock versions because to use two different sets of test cases would be *even more* confusing.  But to run the same test cases in the mock case I need to include a subset of the same yaml from search-api in the mock environment here.  If we can just get past this circular dependency issue the mock mechanism will cease to be necessary, but at this moment for this PR it is needed.  Thus the yaml that duplicates part of search-api is also needed.